src/main/java/com/adobe/target/edge/client/ondevice/collator/PageParamsCollator.java

```diff
@@ -95,26 +96,37 @@ private String extractTopLevel(String host) {
     if (host == null) {
       return "";
     }
-    int idx = host.lastIndexOf('.');
-    if (idx >= 0 && host.length() > 1) {
-      return host.substring(idx + 1);
+    String lowerCaseHost = host.toLowerCase();
+    InternetDomainName publicSuffix = InternetDomainName.from(host).publicSuffix();
+    if (publicSuffix == null) {
+      return "";
+    }
+    String lowerCaseResult = publicSuffix.toString();
+    if (lowerCaseHost.equals(host)) {
+      return lowerCaseResult;
     }
-    return host;
+    return lowerCaseResult.toUpperCase();
   }
 
   private String extractSubDomain(String host) {
-    // TODO: implement properly
     if (host == null) {
       return "";
     }
-    if (host.toLowerCase().startsWith("www.")) {
+    String lowerCaseHost = host.toLowerCase();
+    if (lowerCaseHost.startsWith("www.")) {
+      lowerCaseHost = lowerCaseHost.substring(4);
       host = host.substring(4);
     }
-    String[] parts = host.split("\\.");
-    if (parts.length < 3) {
+    InternetDomainName domain = InternetDomainName.from(lowerCaseHost);
+    if (!domain.hasPublicSuffix()) {
+      return "";
+    }
+    String topPrivateDomain = domain.topPrivateDomain().toString();
+    int endIndex = lowerCaseHost.indexOf(topPrivateDomain);
+    if (endIndex == 0) {
```

Comment:
Is there a test case for this condition?  Trying to think of what host requires this if statement.

Reply:
This condition should apply for all hosts that do not have a subdomain. For example, as we do not treat `www` as a subdomain, the following domain `www.amazon.co.uk` would not have a subdomain. 

<img width="387" alt="image" src="https://user-images.githubusercontent.com/33603621/197068777-3ec8ab64-b8aa-41c1-82f2-57ddc0a71a7b.png">

As the `topPrivateDomain` is `amazon.co.uk`, then this condition would be met